test/servername_test.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+
+#include <openssl/opensslconf.h>
+#include <openssl/bio.h>
+#include <openssl/crypto.h>
+#include <openssl/evp.h>
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+#include <time.h>
+
+#include "../ssl/packet_locl.h"
```

Comment:
Were it not for the use of ```PACKET_``` functions, I think these test cases could go in sslapitest.c.  It may be good to get input from more of the team as to what use of packet_locl.h in tests is acceptable (I've tried to use it in the past and had it rejected).  But for now, leaving this file as a standalone test executable seems fine.